supervision/metrics/intersection_over_union.py

```diff
@@ -0,0 +1,136 @@
+from typing import TYPE_CHECKING, Dict, Union
+
+import numpy as np
+import numpy.typing as npt
+
+from supervision.detection.core import Detections
+from supervision.metrics.core import InternalMetricDataStore, Metric, MetricTarget
+
+if TYPE_CHECKING:
+    pass
+
+
+class IntersectionOverUnion(Metric):
+    def __init__(
+        self,
+        metric_target: MetricTarget = MetricTarget.BOXES,
+        class_agnostic: bool = False,
+        iou_threshold: float = 0.25,
+    ):
+        # TODO: implement for masks and oriented bounding boxes
+        if metric_target in [MetricTarget.MASKS, MetricTarget.ORIENTED_BOUNDING_BOXES]:
+            raise NotImplementedError(
+                f"Intersection over union is not implemented for {metric_target}."
+            )
+
+        self._metric_target = metric_target
+        self._class_agnostic = class_agnostic
+        self._iou_threshold = iou_threshold
+
+        self._store = InternalMetricDataStore(metric_target, class_agnostic)
+
+    def reset(self) -> None:
+        self._store.reset()
+
+    def update(
+        self,
+        data_1: Union[npt.NDArray, Detections],
+        data_2: Union[npt.NDArray, Detections],
+    ) -> Metric:
```

Comment:
Do we return `Metric` or `IntersectionOverUnion` here?

Reply:
Good catch. Should be the latter

Reply:
addressed